lib/traits/device_widget_factory.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:bootstrap_icons/bootstrap_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:yonomi_device_widgets/providers/battery_level_provider.dart';
+import 'package:yonomi_device_widgets/providers/device_provider.dart';
+import 'package:yonomi_device_widgets/providers/lock_provider.dart';
+import 'package:yonomi_device_widgets/providers/power_trait_provider.dart';
+import 'package:yonomi_device_widgets/providers/thermostat_provider.dart';
+import 'package:yonomi_device_widgets/traits/battery_widget.dart';
+import 'package:yonomi_device_widgets/traits/lock_widget.dart';
+import 'package:yonomi_device_widgets/traits/power_widget.dart';
+import 'package:yonomi_device_widgets/traits/slim/base_slim_widget.dart';
+import 'package:yonomi_device_widgets/traits/slim/battery_slim_widget.dart';
+import 'package:yonomi_device_widgets/traits/slim/lock_slim_widget.dart';
+import 'package:yonomi_device_widgets/traits/slim/power_slim_widget.dart';
+import 'package:yonomi_device_widgets/traits/slim/thermostat_slim_widget.dart';
+import 'package:yonomi_device_widgets/traits/slim/unknown_slim_widget.dart';
+import 'package:yonomi_device_widgets/traits/thermostat_widget.dart';
+import 'package:yonomi_device_widgets/traits/unknown_widget.dart';
+import 'package:yonomi_device_widgets/ui/widget_style_constants.dart';
+
+class DeviceWidgetFactory<T> {
+  static Widget produceWidget<T extends DeviceProvider>(
```

Comment:
Can we have T extending Trait instead of provider here, same for the other method as well.